### DISCUSSION TITLE
Airgap Install with Security Onion on Ubuntu 20.04

### DISCUSSION THREAD

**Discussion by rskntroot** [Security-Onion-Solutions/securityonion#6429] [Unsupported Versions] [answered]
We have run into an issue with the installation of SO2.3.90 where certain models of Dell Poweredge Servers (R650 in my case) do not support (or provide RAID array drivers to) CentOS leading to the SO2.3.90 install failing.  In Anaconda, the physical volume simply fails to appear which crashes the installer as it starts.

[Assumption] From what I can tell this seems to be due to a DELL decision to not support CentOS moving forward.  

We were able to successfully install base versions for both RHEL7.9 and Ubuntu 20.04 on the PowerEdge server. Of course, only the Ubuntu 20.04 image was able to run the securityonion-master/so-setup-network script.  The script promptly fails due to the air-gapped nature of the environment.

**We are wondering if there was a Security Onion validated Ubuntu 20.04 image already available for download or if someone has already attempted this.**

Other Potential Solutions: 
1. Move and install all the dependencies for SO2.3.90 manually into the air-gapped environment and attempting to modify the setup scripts, but believe this take an egregious amount of time to sort out.

2. Packing an Ubuntu 20.04 live ISO with a deployment of SO2.3.90 (pre-setup).

3. Repacking the SO2.3.90 ISO with the drivers for the Dell PowerEdge servers, but first we would need to acquire said drivers (if they are even available.)  However, I would not like our team to run into issues with Dell Support with CentOS.

I would appreciate any advice, thank you.

**Comment by dougburks** [comment, marked as answer]
We only support airgap deployments using our Security Onion ISO image based on CentOS 7:
https://docs.securityonion.net/en/2.3/airgap.html

If you are considering purchasing hardware, please consider our appliances as they come pre-loaded with our software and you don't have to worry about driver issues:
https://securityonionsolutions.com/hardware/

Thanks!

**Reply by rskntroot** [reply]
Thank you for the reply.  I was able to get the dependencies migrated into the offline environment and edited the setup script accordingly.  I was able to fully deploy a distributed manager/search/forward cluster on Ubuntu 20.04 with SO2.3.90-Airgap

Data collection is working as expected.  Since it is all in docker I am surprised this isn't support across other OS

**Reply by stevmills** [reply]
@rskntroot do you have a quicklist of the dependencies and script modifications? This is exactly what I need to be able to do! Thanks.

**Reply by rskntroot** [reply]
@stevmills sorry, I just got back from vacation. 
Short list of what I did was first to do a full standalone install on a network with internet access and used the apt-cache after the install to build an offline-repo see: (https://www.guyrutenberg.com/2016/07/15/creating-a-personal-apt-repository-using-dpkg-scanpackages/) 
Did an apt download with those package and then moved them to the offline environment and hosted the lot over https.
Took the docker repo and install scripts from the ISO and put them on the offline ubu20 boxes/vms.
Made changes to the install script (so-functions and so-setup) to prevent any calls to download GPG keys, etc. As well as made sure to enable the air-gap/iso install prompt displayed for each the three deployment types I needed: Distributed Manager, Search, and Forwarder nodes.

It was trial and error editing/running "so-setup iso" until I finally got a full installation to work for the 3 types.  Wish I could just give you more than the theory I was able to get working, sorry!
